### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**
[DEMO LINK](https://dariia-romanova.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,86 @@
...
+  state = {
+    goods: [],
+    isVisible: false,
+    errorMessage: '',
+    isLoading: false,
+  };
+
+  loadGoods = async (callback: () => Promise<Good[]>) => {
+    this.setState({
+      goods: [],
```
Comment (Vitaliy-0) [inline src/App.tsx:24]: why do you need it?

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,86 @@
...
+
+    return (
+      <div className="app">
+        <h1 className="app__title">
+          Dynamic list of Goods
+        </h1>
+
+        {isVisible && <GoodsList goods={goods} isLoading={isLoading} />}
+
+        <div className="app__buttons">
```
Comment (Vitaliy-0) [inline src/App.tsx:54]: will be better to replace buttons under the list

**Review by Vitaliy-0 — APPROVED** [review]
Good job!

File: src/App.tsx (hunk 1 above)
Comment (dariia-romanova) [reply src/App.tsx:24]: I wanted to clear list while loading)

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,86 @@
...
-// or
-// import * as goodsAPI from './api/goods';
+import { GoodsList } from './components/GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[] | [],
```
Comment (AlinaMatuschak) [inline src/App.tsx:8]:
```suggestion
  goods: Good[],
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,12 +1,86 @@
...
+  isVisible: boolean,
+  errorMessage: string,
+  isLoading: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+    isVisible: false,
```
Comment (AlinaMatuschak) [inline src/App.tsx:17]: is this necessary?

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**